File: sketches/sketches.model/src/test/java/edu/kit/kastel/lissa/sketches/model/TestConverts.java
package edu.kit.kastel.lissa.sketches.model;

import static org.junit.Assert.assertTrue;
import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import edu.kit.kastel.lissa.sketches.model.elements.uml.IUMLClass;
import edu.kit.kastel.lissa.sketches.model.elements.uml.IUMLInterface;
import edu.kit.kastel.lissa.sketches.model.impl.Box;
import edu.kit.kastel.lissa.sketches.model.types.SketchBoxTypes;

public class TestConverts {
	private Box boxElement;

	@BeforeEach
	public void setup() {
		this.boxElement = new Box("TestElement", 0.5);
	}

	@Test
	public void convertToClass() {
		IUMLClass clazzRaw = SketchBoxTypes.CLASS.map(this.boxElement, IUMLClass.class);
		assertTrue(clazzRaw instanceof IUMLClass);
	}

	@Test
	public void convertIllegal() {
		assertThrows(IllegalArgumentException.class, () -> SketchBoxTypes.CLASS.map(this.boxElement, IUMLInterface.class));
	}

}

File: sketches/model/src/test/kotlin/edu/kit/kastel/lissa/sketches/model/TestConverts.kt
package edu.kit.kastel.lissa.sketches.model

import edu.kit.kastel.lissa.sketches.model.elements.class_diagram.IClass
import edu.kit.kastel.lissa.sketches.model.elements.generic.IInterface
import edu.kit.kastel.lissa.sketches.model.impl.BoxImpl
import edu.kit.kastel.lissa.sketches.model.types.SketchBoxTypes
import edu.kit.kastel.lissa.sketches.model.types.SketchElementType
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

internal class TestConverts : TestBase() {
    private lateinit var boxElement: BoxImpl

    @BeforeEach
    fun setup() {
        boxElement = BoxImpl("TestElement", 0.5)
        assertEquals(SketchElementType.BOX, boxElement.elementType())
    }

    @Test
    fun convertToClass() {
        val clazzRaw = SketchBoxTypes.CLASS.map(boxElement, IClass::class)
        assertEquals(0.5, clazzRaw.currentConfidence())
        assertEquals("TestElement", clazzRaw.name())
        assertEquals(SketchBoxTypes.CLASS, clazzRaw.currentInterpretation())
        assertEquals(SketchElementType.BOX, clazzRaw.elementType())
    }

    @Test
    fun convertIllegal() {
        Assertions.assertThrows(IllegalArgumentException::class.java) {
            SketchBoxTypes.CLASS.map(boxElement, IInterface::class)
        }
    }
}
